### PR TITLE
Fix #13578: Added custom no of rows

### LINKED ISSUE #13578: Allow custom number of rows to be inserted
### Steps to reproduce
1. Login with username password
2. Select any database
3. Select any table
4. From top navigation select "Insert"

### Expected behaviour
We can add custom number of rows

### Actual behaviour
Bound to insert specific number of rows
enhs

### PR DESCRIPTION
**Pull request by codehimanshu**
Changed select typoe input to number type input.

Signed-off-by: Himanshu Agrawal <himanshuagrawal1998@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

### CODE REVIEW

**Comment by ibennetch** [comment]
It looks like this was covered under the test suite, so now there is a failure about the changed HTML. Could you also fix the test?

```
1) PMA_InsertEditTest::testGetContinueInsertionForm

Failed asserting that '<form id="continueForm" method="post" action="tbl_replace.php" name="continueForm"><input type="hidden" name="db" value="db" /><input type="hidden" name="table" value="tbl" /><input type="hidden" name="lang" value="en" /><input type="hidden" name="token" value="token" /><input type="hidden" name="goto" value="index.php" /><input type="hidden" name="err_url" value="localhost" /><input type="hidden" name="sql_query" value="SELECT 1" /><input type="hidden" name="where_clause[0]" value="a&lt;b" />

Continue insertion with <input type="number" name="insert_rows" id="insert_rows" value="2" min="1" /> rows</form>

' contains "<option value="1" selected="selected">".

/home/travis/build/phpmyadmin/phpmyadmin/test/libraries/PMA_insert_edit_test.php:1483
```

**Comment by codehimanshu** [comment]
Ya sure, will fix this.
Any link or guidelines to setup travis on my local to run tests before making Commit?

**Comment by nijel** [comment]
You do not need Travis to run the tests, see https://github.com/phpmyadmin/phpmyadmin/wiki/Unit_Testing. However to setup Travis you can just go to https://travis-ci.org/ and enable builds on your fork.

File: libraries/insert_edit.lib.php [hunk 1]
```diff
@@ -1453,18 +1453,9 @@ function PMA_getContinueInsertionForm($table, $db, $where_clause_array, $err_url
...
-    $option_values = array(1, 2, 5, 10, 15, 20, 30, 40);
 
-    foreach ($option_values as $value) {
-        $tmp .= '<option value="' . $value . '"';
-        if ($value == $GLOBALS['cfg']['InsertRows']) {
-            $tmp .= ' selected="selected"';
-        }
-        $tmp .= '>' . $value . '</option>' . "\n";
-    }
+    $tmp = '<input type="number" name="insert_rows" id="insert_rows" value="2" min="1" />';
```
Comment (ibennetch) [inline libraries/insert_edit.lib.php]: The value here should reflect the default setting from the [InsertRows](https://docs.phpmyadmin.net/en/latest/config.html#cfg_InsertRows) configuration directive.

**Comment by ibennetch** [comment]
Thanks for your submission!

I like this change, but I have a concern about whether users will know they need to change focus out of the input field for the additional forms to be added to the page. I wonder if it would be possible to update it immediately after the user enters the number rather than requiring them to change focus.

With the current system, if I change the dropdown the number of fields immediately changes; I don't first need to click away (or perhaps the browser changes focus after selecting something from the dropdown, I don't recall offhand how the specifications handle this case). I think users will be looking for an "update" button or expect the change to be immediate.

**Comment by codehimanshu** [comment]
@ibennetch  Changing **on change** to on **keyup/mouseup** seems to do the job. But if I have some data in 2 rows filled, and 2 other rows are empty. And I have to just go and change the number of rows to 2 from 4, I can simply press the down arrow key, but if I clear its content and type 2, all my data of the 2 rows in form is lost. 
Its better to have the change take place **on change** only.
` $(document).on('keyup mouseup', "#insert_rows", function (event) { `

**Comment by codehimanshu** [comment]
@ibennetch any update on this?

File: libraries/insert_edit.lib.php [hunk 2]
```diff
@@ -1453,18 +1453,9 @@ function PMA_getContinueInsertionForm($table, $db, $where_clause_array, $err_url
...
-    $option_values = array(1, 2, 5, 10, 15, 20, 30, 40);
 
-    foreach ($option_values as $value) {
-        $tmp .= '<option value="' . $value . '"';
-        if ($value == $GLOBALS['cfg']['InsertRows']) {
-            $tmp .= ' selected="selected"';
-        }
-        $tmp .= '>' . $value . '</option>' . "\n";
-    }
+    $tmp = '<input type="number" name="insert_rows" id="insert_rows" value="'.$GLOBALS['cfg']['InsertRows'].'" min="1" />';
```
Comment (nijel) [inline libraries/insert_edit.lib.php]: AFAIK we're using `<div class='slider'></div>` is most  places, so this should be consistent with rest of the code.

Comment (codehimanshu) [reply libraries/insert_edit.lib.php]: I didn't get where the class slider is being used! If someone could help out.

**Pull request closed without merging**